LibCarla/source/carla/client/Client.h

```diff
@@ -6,29 +6,13 @@
 
 #pragma once
 
```

Comment:
What about having a client that just follows https://en.cppreference.com/w/cpp/language/pimpl

Then there can be a python client and a c++ client 'implemtnation' and all that users of either interact with is the pimpl layer (which makes including the pimpl layer in a project pretty straightfward).

Reply:
Yes, the pimpl is in the detail::Client, this class is pretty useless by itself, the API depends on all the other classes that are in the "client" folder.

What it would be nice is to have a pure-C interface with the very minimum for communicating with the server, so it would be easier to link anywhere. In practice however the communication is quite complex, a lot of things are cashed to avoid calling the server all the time. It would require some thought doing such interface.